### PR TITLE
Univalence, stage 1.

### PR DESCRIPTION
**Pull request by favonia**
Add more examples.

### CODE REVIEW

**Comment by favonia** [comment]
@jonsterling Look at this short proof:
```
Thm Discrete/reflection {l:lvl} : [
 (->
  [ty : (U l discrete)]
  [a b : ty]
  [p : (path {_} ty a b)]
  (= ty a b))
] by [
  lam ty, a, b, p. elim ty;
  `(coe{0~>1} {x} (= ,ty ,a (@ ,p x)) ax)
].
```

**Comment by jonsterling** [comment]
@favonia WOW! That is so cool.

**Comment by favonia** [comment]
@jonsterling A question: I am trying to prove this goal:
```
  Goal #whatever.
    b : bool
    a : bool
    p : (path {_} bool ($ Not a) b)
    a' : bool
    p' : (path {_} bool ($ Not a') b)
    >> (abs {_} (@ p _))
       =
       (abs {_} (@ p' _))
       in (path {_} bool ($ Not a) b)
```
by
```
fresh i -> refine path/eq/abs;
[ let eq : [(= bool (@ ,p' i) (@ ,p i)) true] = ?. ?
]; auto
```
and got the error `Fail: expected hyp == dim` because, I guess, `i` is not of sort `hyp`. I am not sure why there is such an error.

**Comment by jonsterling** [comment]
@favonia Try using `fresh i:dim -> ...`.

File: test/success/univalence-types.prl [hunk 1]
```diff
@@ -32,28 +32,130 @@ Thm IdIsEquiv{i:lvl} : [
...
+  }
+].
+
+Thm NotIsEquiv : [
+  (IsEquiv bool bool Not)
+] by [
+  unfold IsEquiv; unfold IsContr; unfold Fiber; unfold HasAllPaths;
+  lam b.
+  < proj1 =
+     <proj1 = `($ Not ,b),
```
Comment (jonsterling) [inline test/success/univalence-types.prl]: here, you could write `use Not [use b]`.

Comment (favonia) [reply test/success/univalence-types.prl]: Weird... it raised `Bind`.

**Comment by favonia** [comment]
> Try using fresh i:dim -> ....

Eventually I replaced that proof with something fancier.

File: test/success/univalence-types.prl (hunk 1 above)
Comment (jonsterling) [reply test/success/univalence-types.prl]: weird!

Comment (favonia) [reply test/success/univalence-types.prl]: Is it possible that the `use` tactic relies on knowing the type? `Not` is just a `Def`. In any case, I would like to merge this PR without solving this issue.

File: src/redprl/refiner_types.fun [hunk 2]
```diff
@@ -799,7 +799,7 @@ struct
         val goal1 = makeMem (I, H) (m, (dfun, l, k))
         val goal2 = makeEqIfDifferent (I, H) ((m', n), (dfun, NONE, K.top))
       in
-        |>: goal1 >:? goal2 #> (I, H, trivial)
+        |>:? goal2 >: goal1 #> (I, H, trivial)
```
Comment (jonsterling) [inline src/redprl/refiner_types.fun]: Out of curiosity, how come you prefer this order?

Comment (favonia) [reply src/redprl/refiner_types.fun]: I feel in my experiments `goal2` is the "real" goal. Image you are proving `p = p'` in some path type. The goal with an eta-expanded term is the "real" one, and `goal1` is like some sanity checking.

File: test/success/univalence-types.prl (hunk 1 above)
Comment (jonsterling) [reply test/success/univalence-types.prl]: Oh, you're right!

File: src/redprl/refiner_types.fun (hunk 2 above)
Comment (jonsterling) [reply src/redprl/refiner_types.fun]: ok! Cool.

**Pull request merged**